docs/guides/hooks/index.rst

```diff
@@ -0,0 +1,209 @@
+attr data classes that encapsulate the event data.
+
+.. code-block:: python
+
+    from openedx_events.learning.data import UserData, UserPersonalData
+    from openedx_events.learning.signals import STUDENT_REGISTRATION_COMPLETED
+
+    STUDENT_REGISTRATION_COMPLETED.send_event(
+        user=UserData(
+            pii=UserPersonalData(
+                username=user.username,
+                email=user.email,
+                name=user.profile.name,
+            ),
+            id=user.id,
+            is_active=user.is_active,
+        ),
+    )
+
+You can do this both from the edx-platform code as well as from an openedx
+plugin.
+
+
+Testing events
+^^^^^^^^^^^^^^
+
+Testing your code in CI, specially for plugins is now possible without having to
+import the complete edx-platform as a dependency.
+
+To test your functions you need to include the openedx-events library in your
+testing dependencies and make the signal connection in your test case.
+
+.. code-block:: python
+
+    from openedx_events.learning.signals import STUDENT_REGISTRATION_COMPLETED
+
+    def test_your_receiver(self):
+        STUDENT_REGISTRATION_COMPLETED.connect(your_function)
+        STUDENT_REGISTRATION_COMPLETED.send_event(
+            user=UserData(
+                pii=UserPersonalData(
+                    username='test_username',
+                    email='test_email@example.com',
+                    name='test_name',
+                ),
+                id=1,
+                is_active=True,
+            ),
+        )
+
+        # run your assertions
+
+
+Changes in the openedx-events library that are not compatible with your code
+should break this kind of test in CI and let you know you need to upgrade your
+code.
+
+
+Live example
+^^^^^^^^^^^^
+
+For a complete and detailed example you can see the `openedx-events-2-zapier`_
+plugin. This is a fully functional plugin that connects to
+``STUDENT_REGISTRATION_COMPLETED`` and ``COURSE_ENROLLMENT_CREATED`` and sends
+the relevant information to zapier.com using a webhook.
+
+.. _openedx-events-2-zapier: https://github.com/eduNEXT/openedx-events-2-zapier
+
+
+Index of Events
+-----------------
+
+.. list-table::
+   :widths: 35 50 20
+
+   * - *Name*
+     - *Type*
+     - *Status*
+
+   * - `STUDENT_REGISTRATION_COMPLETED <https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/signals.py#L18>`_
+     - org.openedx.learning.student.registration.completed.v1
+     - Active | `location <https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/views/register.py#L258>`_
+
+   * - `SESSION_LOGIN_COMPLETED <https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/signals.py#L30>`_
+     - org.openedx.learning.auth.session.login.completed.v1
+     - Active | `location <https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/views/login.py#L306>`_
```

Comment:
I did not know this about the rst links. Thanks. I tested running `make technical-docs` and it build in my devstack.

I added a note at the beginning of the list about the two links and changed the 'location' for a date added. This also removes the status since we really don't have a clear expectation of what goes there or what does `active` mean. Instead I'm writing the date the event was added to the platform with the idea that a developer sees it and they can figure if the event has been in the platform for long it is likely more stable.

This idea about the dates is something we could further extend to the annotations in the openedx-events library. The moment an event is created, we can put an `experimentan_until` annotation. This way, even if the original dev does not go back to that repo to say the event is no longer experimental, but fully supported or active a dev looking at this in the future can extrapolate from the date.